homeassistant/components/ted/coordinator.py

```diff
@@ -0,0 +1,68 @@
+"""Coordinator for The Energy Detective (TED) integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+import async_timeout
+import httpx
+from tedpy import TED
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+SCAN_INTERVAL = timedelta(seconds=60)
+TIMEOUT = 50  # 10 seconds less than the scan interval
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TedUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching IPP data from single endpoint."""
+
+    def __init__(self, hass: HomeAssistant, *, name: str, ted_reader: TED) -> None:
+        """Initialize global IPP data updater."""
```

Comment:
Apparently you took the exemple of IPP 😄

Reply:
I made it a bit too obvious, didn't I? Oops.